src/cpp/server/server.cc

```diff
@@ -185,28 +185,35 @@ class Server::SyncRequest GRPC_FINAL : public CompletionQueueTag {
   grpc_completion_queue* cq_;
 };
 
-static grpc_server* CreateServer(int max_message_size) {
+static grpc_server* CreateServer(
+    int max_message_size, const grpc_compression_options& compression_options) {
   if (max_message_size > 0) {
```

Comment:
So you only care about the compression_options when there is a max_message_size set?